workflow/controller/operator.go

```diff
@@ -2628,7 +2628,7 @@ func (woc *wfOperationCtx) getPodByNode(node *wfv1.NodeStatus) (*apiv1.Pod, erro
 		return nil, fmt.Errorf("Expected node type %s, got %s", wfv1.NodeTypePod, node.Type)
 	}
 
-	podName := woc.getPodName(node.Name, node.TemplateName)
+	podName := woc.getPodName(node.Name, wfutil.GetTemplateFromNode(*node))
```

Comment:
we should eventually refactor this to just accept the `node` as an argument. it requires refactoring elsewhere though, similar to #12964 "Future Work"